### PR TITLE
Correct the on-chain script checked in gossip verification

### PR DESCRIPTION
**Pull request by TheBlueMatt**
The `bitcoin_key_1` and `bitcoin_key_2` fields in
`channel_announcement` messages are sorted according to node_ids
rather than the keys themselves, however the on-chain funding
script is sorted according to the bitcoin keys themselves. Thus,
with some probability, we end up checking that the on-chain script
matches the wrong script and rejecting the channel announcement.

The correct solution is to use our existing channel funding script
generation function which ensure we always match what we generate.

This was found in testing the Java bindings, where a test checks
that retunring the generated funding script in `chain::Access`
results in the constructed channel ending up in our network graph.

### CODE REVIEW

**Review by dunxen — COMMENTED** [review]
Looks like `get_signed_channel_announcement()` in the gossip tests makes the same assumption for the bitcoin keys and needs to be updated too.

Otherwise LGTM!

**Review by tnull — COMMENTED** [review]
Generally looks good. Could you maybe add a check to some test case to ensure we catch this (or similar) issue not only down the line in bindings generation, if there ever were to be a regression?

File: lightning/src/routing/gossip.rs [hunk 1]
```diff
@@ -1455,11 +1456,8 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 			&Some(ref chain_access) => {
 				match chain_access.get_utxo(&msg.chain_hash, msg.short_channel_id) {
 					Ok(TxOut { value, script_pubkey }) => {
-						let expected_script = Builder::new().push_opcode(opcodes::all::OP_PUSHNUM_2)
```
Comment (tnull) [inline lightning/src/routing/gossip.rs:1458]: Seems the `Builder` and `opcodes` imports can be removed now.

**Review by valentinewallace — COMMENTED** [review]
Looks good after outstanding feedback and CI

**Comment by tnull** [comment]
> Pushed a fix, as for a check for this, I'm not entirely sure how to write one that is actually useful - sure I could write a test that checks the specific scripts which failed here, but that's not all that interesting, sadly we had several checks which ultimately should have caught this, but of course we replicated the same bug into the tests, which made them not all that helpful.

Alright, then hopefully tests will just catch if there ever would be a regression. Seems the fix didn't do the trick entirely though.

**Comment by valentinewallace** [comment]
Seems we're still getting a test panic. With the amount of test panics that arose, I'm happy with the coverage 😆

**Review by dunxen — DISMISSED** [review]
ACK 43814f3

Back to the nominal crate test failures for those two lol

**Review by tnull — DISMISSED** [review]

**Review by dunxen — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Pull request merged**